src/rpm/header.go

```diff
@@ -0,0 +1,180 @@
+		{ID: tagFileFlags, Type: rpmTypeInt32, Value: filesInfo.FileFlags},
+		{ID: tagFileLangs, Type: rpmTypeStringArray, Value: filesInfo.FileLangs},
+		{ID: tagFileDigests, Type: rpmTypeStringArray, Value: filesInfo.FileDigests},
+		{ID: tagFileLinkTos, Type: rpmTypeStringArray, Value: filesInfo.FileLinkTos},
+
+		{ID: tagSize, Type: rpmTypeInt32, Value: []int32{int32(payloadSize)}},
+		{ID: tagPayloadDigest, Type: rpmTypeStringArray, Value: []string{payloadDigest}},
+		{ID: tagPayloadDigestAlgo, Type: rpmTypeInt32, Value: []int32{int32(payloadDigestAlgo)}},
+	}...)
+
+	if !projectCtx.TarantoolIsEnterprise {
+		// add Tarantool dependency
+		tarantoolDepName := "tarantool"
+		flagGreaterOrEqual := int32(rpmSenseGreater | rpmSenseEqual)
+
+		minVersion := strings.SplitN(projectCtx.TarantoolVersion, "-", 2)[0]
+		maxVersion, err := common.GetNextMajorVersion(minVersion) // XXX: add to project Ctx
+		if err != nil {
+			return nil, fmt.Errorf("Failed to get next major version of Tarantool %s", err)
+		}
+
+		rmpHeader.addTags([]rpmTagType{
+			{ID: tagRequireName, Type: rpmTypeStringArray,
+				Value: []string{tarantoolDepName, tarantoolDepName}},
+			{ID: tagRequireFlags, Type: rpmTypeInt32,
+				Value: []int32{flagGreaterOrEqual, int32(rpmSenseLess)}},
+			{ID: tagRequireVersion, Type: rpmTypeStringArray,
+				Value: []string{minVersion, maxVersion}},
+		}...)
+	}
+
+	return rmpHeader, nil
+}
+
+func getFilesInfo(relPaths []string, dirPath string) (filesInfoType, error) {
+	filesInfo := filesInfoType{}
+
+	for _, relPath := range relPaths {
+		fullFilePath := filepath.Join(dirPath, relPath)
+		fileInfo, err := os.Stat(fullFilePath)
+		if err != nil {
+			return filesInfo, err
+		}
+
+		if fileInfo.Mode().IsRegular() {
+			filesInfo.FileFlags = append(filesInfo.FileFlags, fileFlag) // XXX
+
+			fileDigest, err := common.FileMD5Hex(fullFilePath)
+			if err != nil {
+				return filesInfo, fmt.Errorf("Failed to get file MD5 hex: %s", err)
+			}
+
+			filesInfo.FileDigests = append(filesInfo.FileDigests, fileDigest)
+		} else {
+			filesInfo.FileFlags = append(filesInfo.FileFlags, dirFlag) // XXX
+			filesInfo.FileDigests = append(filesInfo.FileDigests, emptyDigest)
+		}
+
+		fileDir := filepath.Dir(relPath)
+		fileDir = fmt.Sprintf("/%s/", fileDir)
+		dirIndex := addDirAngGetIndex(&filesInfo.DirNames, fileDir)
+		filesInfo.DirIndexes = append(filesInfo.DirIndexes, int32(dirIndex))
+
+		filesInfo.BaseNames = append(filesInfo.BaseNames, filepath.Base(relPath))
+		filesInfo.FileMtimes = append(filesInfo.FileMtimes, int32(fileInfo.ModTime().Unix()))
+
+		filesInfo.FileUserNames = append(filesInfo.FileUserNames, defaultFileUser)
+		filesInfo.FileGroupNames = append(filesInfo.FileGroupNames, defaultFileGroup)
+		filesInfo.FileLangs = append(filesInfo.FileLangs, defaultFileLang)
+		filesInfo.FileLinkTos = append(filesInfo.FileLinkTos, defaultFileLinkTo)
+
+		sysFileInfo, ok := fileInfo.Sys().(*syscall.Stat_t)
```

Comment:
`syscall.Stat_t` does it really needed?

Aren't any "high-level" functions for that?

Reply:
I didn't find another way to get file inode.

Reply:
Not sure, check please - https://stackoverflow.com/a/28339521

Reply:
But it uses `syscall.Stat_t` too...

Reply:
without ugly `(*syscall.Stat_t)` cast

Reply:
Hmmm... But ok... I've missed something